rest-api/dao/participant_summary_dao.py

```diff
@@ -336,6 +395,69 @@ def calculate_enrollment_status(self, consent_for_study_enrollment_and_ehr,
       return EnrollmentStatus.MEMBER
     return EnrollmentStatus.INTERESTED
 
+  def calculate_member_time(self, consent_for_study_enrollment_and_ehr, participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.enrollmentStatusMemberTime is not None:
+      return participant_summary.enrollmentStatusMemberTime
+    elif consent_for_study_enrollment_and_ehr:
+      return participant_summary.consentForElectronicHealthRecordsTime
+    else:
+      return None
+
+  def calculate_core_stored_sample_time(self, consent_for_study_enrollment_and_ehr,
+                                        participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.numCompletedBaselinePPIModules == \
+      self._get_num_baseline_ppi_modules() and \
+      participant_summary.physicalMeasurementsStatus == PhysicalMeasurementsStatus.COMPLETED and \
+      participant_summary.samplesToIsolateDNA == SampleStatus.RECEIVED:
+
+      max_core_sample_time = self.calculate_max_core_sample_time(participant_summary,
+                                                                 field_name_prefix='sampleStatus')
+
+      if max_core_sample_time and participant_summary.enrollmentStatusCoreStoredSampleTime:
+        return participant_summary.enrollmentStatusCoreStoredSampleTime
+      else:
+        return max_core_sample_time
+    else:
+      return None
+
+  def calculate_core_ordered_sample_time(self, consent_for_study_enrollment_and_ehr,
+                                         participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.numCompletedBaselinePPIModules == \
+      self._get_num_baseline_ppi_modules() and \
+      participant_summary.physicalMeasurementsStatus == PhysicalMeasurementsStatus.COMPLETED:
+
+      max_core_sample_time = self.calculate_max_core_sample_time(
+        participant_summary, field_name_prefix='sampleOrderStatus')
+
+      if max_core_sample_time and participant_summary.enrollmentStatusCoreOrderedSampleTime:
+        return participant_summary.enrollmentStatusCoreOrderedSampleTime
+      else:
+        return max_core_sample_time
```

Comment:
change the logic here, if max_core_sample_time is not None, return the existing value.